src/backends/torch/torchlib.cc

```diff
@@ -0,0 +1,142 @@
+    {
+        this->_libname = "torch";
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::TorchLib(TorchLib &&tl) noexcept
+        : MLLib<TInputConnectorStrategy,TOutputConnectorStrategy,TorchModel>(std::move(tl))
+    {
+        this->_libname = "torch";
+        _traced = std::move(tl._traced);
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::~TorchLib() 
+    {
+        
+    }
+
+    /*- from mllib -*/
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    void TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::init_mllib(const APIData &ad) 
+    {
+        bool gpu = false;
+        APIData lib_ad = ad.getobj("parameters").getobj("mllib");
+
+        if (lib_ad.has("gpu")) {
+            gpu = lib_ad.get("gpu").get<bool>() && torch::cuda::is_available();
+        }
+        if (lib_ad.has("nclasses")) {
+            _nclasses = lib_ad.get("nclasses").get<int>();
+        }
+
+        _device = gpu ? torch::Device("cuda") : torch::Device("cpu");
+
+        _traced = torch::jit::load(this->_mlmodel._model_file);
+        _traced->to(_device);
+        _traced->eval();
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    void TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::clear_mllib(const APIData &ad) 
+    {
+
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    int TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::train(const APIData &ad, APIData &out) 
+    {
+        
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    int TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::predict(const APIData &ad, APIData &out) 
+    {
+        APIData params = ad.getobj("parameters");
+        APIData output_params = params.getobj("output");
+
+        TInputConnectorStrategy inputc(this->_inputc);
+        TOutputConnectorStrategy outputc;
+        try {
+            inputc.transform(ad);
+        } catch (...) {
+            throw;
+        }
+
+        Tensor output = _traced->forward({inputc._in.to(_device)}).toTensor();
+        output = torch::softmax(output, 1);
+        
+        // Output
+        std::vector<APIData> results_ads;
+
+        for (int i = 0; i < output.size(0); ++i) {
+            std::tuple<Tensor, Tensor> sorted_output = output.slice(0, i, i + 1).sort(1, true);
+
+            APIData results_ad;
+            std::vector<double> probs;
+            std::vector<std::string> cats;
+
+            if (output_params.has("best")) {
```

Comment:
You should be able to put the test on `best` outside of the loop since it bears the same value for every output.